SecretManagement.PleasantPasswordServer.Extension/Private/Invoke-LoginToPleasant.ps1

```diff
@@ -34,6 +34,20 @@ function Invoke-LoginToPleasant
         $AdditionalParameters
     )
 
+    # Check if a login token already exists and is not expired
+    if ($null -ne $script:LoginToken){
```

Comment:
Why do you need the script scope. It is only used inside of this function

Reply:
I have used the script scope to persist the variable between function calls. Without it the variable does not persist and a new login token is generated each time an action is performed on the vault.

Reply:
I don't like the $script scope that much. Just my opinion. We could replace this with `Export-CliXml` the whole token object to $env:Temp token.xml. The same place where your credential.xml file is stored.

If the function gets called you can import the file and check if the token is expired. If It is acquire a new one and replace the token xml-file. Would that be a good idea?